synapse/api/auth/msc3861_delegated.py

```diff
@@ -243,6 +243,24 @@ async def get_user_by_req(
             # so that we don't provision the user if they don't have enough permission:
             requester = await self.get_user_by_access_token(access_token, allow_expired)
 
+            # Allow impersonation by an admin user using `user_id` query parameter
+            if request.args is not None:
+                user_id_params = request.args.get(b"user_id")
+                if user_id_params:
+                    if SCOPE_SYNAPSE_ADMIN in requester.scope:
```

Comment:
```suggestion
                    if self.is_server_admin(requester):
```

If we have an API for it we may as well use it